mllib/src/main/scala/org/apache/spark/ml/classification/OneVsRest.scala

```diff
@@ -163,9 +165,7 @@ final class OneVsRestModel private[ml] (
     val initUDF = udf { () => Map[Int, Double]() }
     val newDataset = dataset.withColumn(accColName, initUDF())
 
-    // persist if underlying dataset is not persistent.
-    val handlePersistence = dataset.rdd.getStorageLevel == StorageLevel.NONE
-    if (handlePersistence) {
+    if ($(handlePersistence)) {
```

Comment:
See comment above, we should also check that `dataset.storageLevel == StorageLevel.NONE` before caching `newDataset`